### PR TITLE
Fix headers with named collections, add compression_method

### LINKED ISSUE #35269: named_collections s3/url do not support compression_method
```sql
cat /etc/clickhouse-server/config.d/s3.xml 
<yandex>
     <named_collections>
        <s3_mydata>
		    <format>CSV</format>
		    <compression_method>gzip</compression_method>
        </s3_mydata>
    </named_collections>
</yandex>


select * from s3(s3_mydata, url = 'https://s3.us-east-1.amazonaws.com/test/test_file.tsv.gz', format = 'RawBLOB') limit 10;


Received exception from server (version 22.2.3):
Code: 42. DB::Exception: Received from localhost:9000. 
DB::Exception: Unknown key-value argument `compression_method` for StorageS3, expected: url, [access_key_id, secret_access_key], name of used format, structure and [compression_method].. 
(NUMBER_OF_ARGUMENTS_DOESNT_MATCH)
```

### LINKED ISSUE #35273: named_collections does not support HTTP headers for `url` engine
I see the test https://github.com/ClickHouse/ClickHouse/blob/master/tests/integration/test_redirect_url_storage/configs/named_collections.xml

but I cannot make it to work

```sql
cat /etc/clickhouse-server/config.d/url.xml

<yandex>
     <named_collections>
        <url_mydata>
	      <headers>
                  <header>
                      <name>X-ClickHouse-Format</name>
                      <value>JSONEachRow</value>
                  </header>
              </headers>
        </url_mydata>
    </named_collections>
</yandex>


select * from url(url_mydata, url='http://127.0.0.1:8123?query=select+today(),now()', format='RawBLOB');
┌─raw_blob──────────────────────────┐
│ 2022-03-14	2022-03-14 17:50:01                 ---------<<<-- X-ClickHouse-Format = JSONEachRow is not applied
 │
└───────────────────────────────────┘
```

maybe headers only apply to POST ?

### PR DESCRIPTION
**Pull request by kssenii**
Changelog category (leave one):
- Bug Fix (user-visible misbehaviour in official stable or prestable release)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix headers with named collections, add compression_method. Closes https://github.com/ClickHouse/ClickHouse/issues/35273. Closes https://github.com/ClickHouse/ClickHouse/issues/35269.

### CODE REVIEW

File: tests/queries/0_stateless/02244_url_engine_headers_test.python [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        print("Fetched answer :", ch_answer, file=sys.stderr)
+        raise Exception("Fail on query")
+
+class HttpProcessor(SimpleHTTPRequestHandler):
+    def _set_headers(self):
+        test_header = self.headers.get('test_header')
+        if test_header and test_header.startswith('test_header_clickhouse'):
+            self.send_response(200)
+        else:
+            self.send_response(403)
```
Comment (kssenii) [inline tests/queries/0_stateless/02244_url_engine_headers_test.python:88]: actual check is here

Comment (tavplubix) [reply tests/queries/0_stateless/02244_url_engine_headers_test.python:88]: So this test only checks that headers are passed, right? Tests looks too complicated for such simple check, maybe just use the query from den-crane's example? I mean `select * from url(url_mydata, url='http://127.0.0.1:8123?query=select+today(),now()', format='RawBLOB');` or something like that

**Review by tavplubix — APPROVED** [review]

**Comment by tavplubix** [comment]
I suppose that 02226_s3_with_cache.sql was added accidentally, wasn't it?

**Comment by kssenii** [comment]
No related test failures:
```
01383_log_broken_table
01037_polygon_dicts_correctness_all
01514_parallel_formatting
```

**Pull request merged**